README.md

```diff
@@ -119,6 +119,184 @@ See github documentation for more information:
+```
+├── build
+│   └── linux
+│       └── x86_64                                  # plugins missing from this directory will be fetched from s3
+│           ├── {snap-plugin-collector-plugin_name} # make will build the current plugin for testing
+│           └── {snap-plugin-publisher-custom}      # custom plugins can be copied here to take precedence over s3 binaries
+├── examples
+│   └── tasks
+│       ├── {example.json}                          # json tasks are ignored and serve purely as examples
+│       └── {example.yml}                           # yaml tasks are loaded and verified in the test framework
+└── scripts
+    ├── large.sh
+    └── test
+        ├── {docker-compose.yml}                    # to be supplied by the developer
+        ├── large_spec.rb                           # default large test, can be disabled by excluding from pluginsync for complex test scenarios
+        ├── {custom_spec.rb}                        # additional spec tests can be written to compliment the default large test
+        ├── {setup.rb}                              # optional setup executed before running large test
+        └── spec_helper.rb
+```
+
+#### Docker Compose
+
+A default `docker_compose.yml` file should be supplied by the developer and placed in `./scripts/test` directory. This will be used by the default large spec test. Additional docker compose config files can be supplied for complex test scenarios and they require their own custom_spec.rb test.
+
+Currently these environment variables are passed to the Snap container (we are investigating ways to pass additional/arbitrary ENV variable values):
+* OS: any os available in [snap-docker repo](https://github.com/intelsdi-x/snap-docker) (default: alpine)
+* SNAP_VERSION: any snap version, or git sha1 that's available in the s3 bucket (default: latest)
+* PLUGIN_PATH: used by large test framework, this must be included in the snap container
+
+Single container:
+```
+version: '2'
+services:
+   snap:                              # NOTE: do not change the snap container name
+    image: intelsdi/snap:${OS}_test
+    environment:
+      SNAP_VERSION: "${SNAP_VERSION}"
+    volumes:
+      - "${PLUGIN_PATH}:/plugin"
+```
+
+Multiple container:
+```
+version: '2'
+services:
+  snap:                               # NOTE: do not change the snap container name
+    image: intelsdi/snap:alpine_test  # OS can be locked down to a specific version
+    environment:
+      SNAP_VERSION: "${SNAP_VERSION}"
+    volumes:
+      - "${PLUGIN_PATH}:/plugin"
+    links:
+      - influxdb
+  influxdb:
+    image: influxdb:1.0
+    expose:
+      - "8083"
+      - "8086"
+```
+
+#### Example Tasks
+
+The default `large_spec.rb` test will verify all yaml example tasks in the plugin's `examples/task` folder. The large test ignores json tasks, so they can be used as examples to demonstrate features that are not tested. There are no restrictions when you write a custom tests, and here's some recommendations:
+
```

Comment:
I would add another recommendation:
* use environment variable in task manifest to set host address (e.g. `host: "${INFLUXDB_HOST}"`). Set that environment variable in `docker-compose.yml`, after SNAP_VERSION (e.g. `INFLUXDB_HOST: influxdb`).